libs/src/api/entityManager.ts

```diff
@@ -148,38 +147,19 @@ export class EntityManager extends Base {
         is_private: playlist.is_private
       }
 
-      if (storageV2UploadEnabled) {
-        const metadataCid = await Utils.fileHasher.generateMetadataCidV1(
-          metadata
-        )
-        if (this.contracts.EntityManagerClient === undefined) {
-          throw new Error('EntityManagerClient is undefined')
-        }
-        const { txReceipt } =
-          await this.contracts.EntityManagerClient.manageEntity(
-            userId,
-            EntityType.PLAYLIST,
-            playlist.playlist_id,
-            Action.CREATE,
-            JSON.stringify({ cid: metadataCid.toString(), data: metadata })
-          )
-        responseValues.blockHash = txReceipt.blockHash
-        responseValues.blockNumber = txReceipt.blockNumber
-        return responseValues
-      } else {
-        const { metadataMultihash } =
-          await this.creatorNode.uploadPlaylistMetadata(metadata)
-        const entityManagerMetadata = writeMetadataThroughChain
-          ? JSON.stringify({ cid: metadataMultihash, data: metadata })
-          : metadataMultihash
-        return await this.manageEntity({
-          userId: userId,
-          entityType,
-          entityId: playlist.playlist_id,
-          action: createAction,
-          metadata: entityManagerMetadata
-        })
-      }
+      this.creatorNode.validatePlaylistSchema(metadata)
+      const metadataCid = await Utils.fileHasher.generateMetadataCidV1(metadata)
+      const entityManagerMetadata = JSON.stringify({
+        cid: metadataCid.toString(),
+        data: metadata
+      })
+      return await this.manageEntity({
```

Comment:
wouldn't this need to first destructure txReceipt and then return an object with the receipt's blockHash and blockNumber?

Reply:
this.manageEntity wraps that, see https://github.com/AudiusProject/audius-protocol/blob/e8bceecd7c7267caa11ad7c17f68ddca16fb0edb/libs/src/api/entityManager.ts#L304

Reply:
ah I see, it's a different manageEntity calling EntityManagerClient.manageEntity